src/libcollections/vec.rs

```diff
@@ -1296,6 +1296,17 @@ impl<T: PartialEq> Vec<T> {
     pub fn dedup(&mut self) {
         self.dedup_by(|a, b| a == b)
     }
+    
+    #[stable(feature = "rust1", since = "1.0.0")]
+    pub fn remove_item(&mut self, item: &T) -> Option<usize> {
+        let pos = match self.iter().position(|x| *x == *item) {
+            Some(x) => x,
+            None => return None,
+        };
+
+        self.remove(pos);
```

Comment:
Perhaps return type should be `Option<(usize, T)>`, returning `Some((pos, self.remove(pos)))`